### PR TITLE
Ensure FileDataProvider is the default provider

### PR DESCRIPTION
**Pull request by franckverrot**
# TL;DR

I noticed another warning that appears only at runtime (I'm using `i18n_data` thru `countries` within `ActiveValidator`). In order to remove it I added a test case on the `data_provider` method as it's responsible for the warning.

# Full explanation
The rewrite of the `data_provider` setter removes a warning that Ruby
detects during execution (when Ruby is executed with the `-w flag`) when we
try to read the `@data_provider` instance variable before initializing it.

This is also why that code couldn't have been written `@data_provider || ...`
because it would have generated yet another warning.

One viable solution is to use the `||=` operator. The reason why it won't
complain is pretty much a side-effect of how Ruby generates the underlying
[bytecode][compile] for the `||=` operator when the left-hand side is `nil`.

Using `||=` comes with a new behavior but there is no guarantee in API,
as of today, that data providers can't be cached.

[compile]: https://github.com/ruby/ruby/blob/7790f37efdd8dd42a0a43c3206f6afdd43f8e86a/compile.c#L4410-L4418

### CODE REVIEW

**Comment by franckverrot** [comment]
@grosser Sorry I'm making you go thru another PR :-(

**Comment by grosser** [comment]
looks good, also gets rid of an unnecessary require :)

**Comment by grosser** [comment]
0.6.2
